Reaktoro/Equilibrium/SmartEquilibriumSolver.cpp

```diff
@@ -277,15 +277,14 @@ struct SmartEquilibriumSolver::Impl
         u = properties.chemicalPotentials();
 
         // Auxiliary references to the derivatives dn/db, dn/dT, dn/dP, and du/dn
-        const auto& dndb = solver.sensitivity().dndb;
-        const auto& dndT = solver.sensitivity().dndT;
-        const auto& dndP = solver.sensitivity().dndP;
-        const auto& dudn = u.ddn;
+        const auto& dndT = sensitivity.dndT;
+        const auto& dndP = sensitivity.dndP;
+        const auto& dndb = sensitivity.dndb;
 
-        // Compute the matrix du/db = du/dn * dn/db
-        dudb = dudn * dndb;
-        dudT = dudn * dndT;
-        dudP = dudn * dndP;
+        // Compute the matrices du/dT, du/dP, du/db
+        dudT = u.ddn * dndT + u.ddT; // du/dT = ∂u/∂n*∂n/∂b + ∂u/∂T
+        dudP = u.ddn * dndP + u.ddP; // du/dP = ∂u/∂n*∂n/∂b + ∂u/∂P
+        dudb = u.ddn * dndb;         // du/du = ∂u/∂n*∂n/∂b
```

Comment:
Note here that we were to take into account the partial derivatives `∂u/∂T` and `∂u/∂P` as well. 
We are computing total derivatives above - we need to add the contribution of `∂u/∂T` and `∂u/∂P`.

This is because of the following functional dependency:

`u = u(T, P, n(T, P, b))`